engineapi/engineapi/contracts_actions.py

```diff
@@ -30,66 +37,136 @@ class InvalidAddressFormat(Exception):
     """
 
 
+class UnsupportedCallRequestType(Exception):
+    """
+    Raised when unsupported call request type specified.
+    """
+
+
+class UnsupportedBlockchain(Exception):
+    """
+    Raised when unsupported blockchain specified.
+    """
+
+
+class CallRequestMethodValueError(Exception):
+    """
+    Raised when method not acceptable for specified request type.
+    """
+
+
+class CallRequestRequiredParamsValueError(Exception):
+    """
+    Raised when required params not acceptable for specified request type.
+    """
+
+
 class ContractAlreadyRegistered(Exception):
     pass
 
 
+def parse_registered_contract_response(
+    obj: Tuple[RegisteredContract, Blockchain]
+) -> data.RegisteredContractResponse:
+    return data.RegisteredContractResponse(
+        id=obj[0].id,
+        blockchain=obj[1].name,
+        address=obj[0].address,
+        metatx_requester_id=obj[0].metatx_requester_id,
+        title=obj[0].title,
+        description=obj[0].description,
+        image_uri=obj[0].image_uri,
+        created_at=obj[0].created_at,
+        updated_at=obj[0].updated_at,
+    )
+
+
+def parse_call_request_response(
+    obj: Tuple[CallRequest, RegisteredContract, CallRequestType, CallRequestType]
+) -> data.CallRequestResponse:
+    return data.CallRequestResponse(
+        id=obj[0].id,
+        contract_id=obj[0].registered_contract_id,
+        contract_address=obj[1].address,
+        metatx_requester_id=obj[0].metatx_requester_id,
+        call_request_type=obj[2].request_type,
+        caller=obj[0].caller,
+        method=obj[0].method,
+        parameters=obj[0].parameters,
+        expires_at=obj[0].expires_at,
+        created_at=obj[0].created_at,
+        updated_at=obj[0].updated_at,
+    )
+
+
 def validate_method_and_params(
-    contract_type: ContractType, method: str, parameters: Dict[str, Any]
-) -> None:
+    call_request_type: str,
```

Comment:
Let's make `call_request_type: Optional[str]`.

If `call_request_type is None`, then let us use `dropper-v0.2.0` as the default request type.

This will allow us to roll out these updates without breaking experience for our existing users - all of them are using `dropper-v0.2.0`.